app/services/foreman/plugin.rb

```diff
@@ -234,15 +238,30 @@ def permission(name, hash, options = {})
 
     # Add a new role if it doesn't exist
     def role(name, permissions)
-      @default_roles[name] = permissions
+      default_roles[name] = permissions
       return false if pending_migrations || Rails.env.test?
 
       Role.transaction do
         role = Role.where(:name => name).first_or_create
         role.add_permissions!(permissions) if role.permissions.empty?
+        rbac_registry.role_ids << role.id
       end
     end
 
+    # Add plugin permissions to core's Manager and Viewer roles
+    # Usage:
+    # add_permissions_to_default_roles ['MyPlugin::FirstResource', 'MyPlugin::SecondResource'], :except => [:skip_this_permission]
+    def add_resource_permissions_to_default_roles(resources, opts = {})
```

Comment:
If we have all permissions registered, why do we need to specify resources here? don't we know these already? it seems weird to me that I basically need to list all classes of my plugin?